src/nnet3/nnet-utils.cc

```diff
@@ -101,9 +280,20 @@ static void ComputeSimpleNnetContextForShift(
     input.indexes.push_back(Index(n, t));
     output.indexes.push_back(Index(n, t));
   }
-  // the assumption here is that the network just requires the ivector at time
-  // t=0.
-  ivector.indexes.push_back(Index(n, 0));
+  // push the indexes for ivector(s)
+  int32 ivector_period = GetInputInterval(nnet,"ivector");
+  if (ivector_period == 0)
+    // single ivector for the entire chunk
+    // the assumption here is that the network just requires the ivector at time
+    // t=0
+    ivector.indexes.push_back(Index(n, 0));
+  else {
+    // multiple ivectors, so push multiple indexes
+    int32 t_begin = std::floor(1.0 * input_start / ivector_period);
```

Comment:
you should be using a method more similar to this to work out the indexes for which you want iVectors in the decoding.
However, it's not good practice to use floating-point arithmetic to compute these quantities-- it doesn't scale well to large values.  You could use (input_start >= 0 ? input_start / ivector_period : (input_start + ivector_period - 1) / ivector_period).  In fact, defining a function
/// Returns a / b, rounding towards negative infinity in all cases.
int32 DivideRoundingDown(int32 a, int32 b);
would be a more elegant way to do this.  You can put this in kaldi-math.h.